state.go

```diff
@@ -357,8 +357,12 @@ func (uv *UpdateVerifyState) Handle(ctx *StateContext, c Controller) (State, boo
 	}
 
 	if has {
-		if uv.update.ArtifactName() == c.GetCurrentArtifactName() {
-			log.Infof("successfully running with new image %v", c.GetCurrentArtifactName())
+		artifactName, err := c.GetCurrentArtifactName()
+		if err != nil {
+			log.Errorf("Cannot determine current artifact. Update will continue anyways: %v : %v", defaultDeviceTypeFile, err)
+		}
```

Comment:
Actually, here we should throw an error. The reason is that this is the verification code before committing a new update, so this is a good place to treat it like an error, roll back, and force the creator of the artifact to fix the problem.

Above you can see how to treat errors in the state machine, and you can also ask Marcin.